dataset/dbengine.py

```diff
@@ -98,22 +72,81 @@ def execute_query(self, query):
 
     def create_db_table_from_query(self, name, query):
         tic = time.clock()
-        drop = drop_table_template.substitute(tab_name=name)
-        create = create_table_template.substitute(tab_name=name, stmt=query)
+        drop = drop_table_template.substitute(table=name)
+        create = create_table_template.substitute(table=name, stmt=query)
         conn = self.engine.connect()
-        dropped = conn.execute(drop)
-        created = conn.execute(create)
+        conn.execute(drop)
+        conn.execute(create)
         conn.close()
         toc = time.clock()
         logging.debug('Time to create table: %.2f secs', toc-tic)
         return True
 
     def create_db_index(self, name, table, attr_list):
-        stmt = index_template.substitute(idx_title=name, table=table, attr=','.join(attr_list))
+        """
+        create_db_index creates a (multi-column) index on the columns/attributes
+        specified in :param attr_list: with the given :param name: on
+        :param table:.
+
+        :param name: (str) name of index
+        :param table: (str) name of table
+        :param attr_list: (list[str]) list of attributes/columns to create index on
+        """
+        # We need to quote each attribute since Postgres auto-downcases unquoted column references
+        quoted_attrs = map(lambda attr: '"{}"'.format(attr), attr_list)
+        stmt = index_template.substitute(idx_title=name, table=table, attrs=','.join(quoted_attrs))
         tic = time.clock()
         conn = self.engine.connect()
         result = conn.execute(stmt)
         conn.close()
         toc = time.clock()
         logging.debug('Time to create index: %.2f secs', toc-tic)
         return result
+
+    def _apply_func(self, func, collection):
+        if self._pool is None:
```

Comment:
Why set self._pool to None when pool size is 1 instead of 1? Is it to avoid overhead incurred by Python multiprocessing library?